core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableValues.java

```diff
@@ -70,6 +74,32 @@ public static EnumerableValues create(RelOptCluster cluster,
     return new EnumerableValues(getCluster(), rowType, tuples, traitSet);
   }
 
+  @Override public RelNode passThrough(final RelTraitSet required) {
+    RelCollation collation = required.getCollation();
+    if (collation == null || collation.isDefault()) {
+      return null;
+    }
+    if (tuples.size() > 1) {
+      Ordering<List<RexLiteral>> ordering = null;
+      for (RelFieldCollation fc : collation.getFieldCollations()) {
+        Ordering<List<RexLiteral>> comparator = RelMdCollation.comparator(fc);
+        if (ordering == null) {
+          ordering = comparator;
+        } else {
+          ordering = ordering.compound(comparator);
+        }
```

Comment:
Can you add comments to describe what this for loop is doing? Sorry but I had a hard time to understand it.

Reply:
Sure, I added comments.